client/state/sites/selectors.js

```diff
@@ -68,36 +67,41 @@ export const getSiteBySlug = createSelector(
 );
 
 /**
- * Returns a normalized site object by its ID. Intends to replicate
- * the site object returned from the legacy `sites-list` module.
- *
+ * Returns a normalized site object by its ID or site slug.
```

Comment:
maybe it's worth mentioning here at the top in the comment that this function depends on state in the selector module